### PR TITLE
Move the backend to /back/

### PR DESCRIPTION
**Pull request by neongreen**
I've also done some cleanup here - removed `b` and the useless changelog, fixed a Cabal warning, etc.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,8 +37,10 @@ tags
 # backend config
 /config.json
 
+# backend JavaScript artifacts
+/back/guidejs/node_modules/
+/back/guidejs/package-lock.json
+/back/static/js/
```
Comment (willbasky) [inline .gitignore:43]: Does backend generates this js code?

File: .travis.yml [hunk 2]
```diff
@@ -19,11 +19,12 @@ cache:
   directories:
     - $HOME/.stack
     - .stack-work
+    - back/.stack-work
```
Comment (willbasky) [inline .travis.yml:25]: Does  `stack-work` stored in both directories?

File: INSTALL.md [hunk 3]
```diff
@@ -2,10 +2,9 @@
 
 The `state/` directory contains the database. You can download the current database of [guide.aelve.com](http://guide.aelve.com) by doing this:
 
-    $ git clone https://github.com/aelve/guide-database
-    $ mv guide-database state
+    $ scripts/official.sh
 
-The `config.json` file contains the config (it will be created at the 1st start). There are 5 settings so far:
+The `config.json` file contains the config (it will be created at the first start):
```
Comment (willbasky) [inline INSTALL.md]: Is there script creating `config.json` or piece of code?

File: README.md [hunk 4]
```diff
@@ -34,34 +34,7 @@ $ java -jar /usr/share/selenium-server/selenium-server-standalone.jar
...
-  * `lib` – actual code (as a library)
-  * `src` – just a dumb `Main.hs` file to compile an executable
-  * `tests` – tests
-  * `static` – icons, CSS, Markdown pages, HTML pages, and some JS libraries
-  * `templates` – HTML templates for pages and elements of pages
-  * `scripts` – some scripts used by automatic testing
-  * `favicon` – code used to generate a favicon
-  * `guidejs` – client side JavaScript
-  * `front` – the new frontend, written in Vue.js
-
```
Comment (willbasky) [inline README.md:50]: Is anywhere frontend structure else?

File: .gitignore (hunk 1 above)
Comment (neongreen) [reply .gitignore:43]: Yep.

File: .travis.yml (hunk 2 above)
Comment (neongreen) [reply .travis.yml:25]: Yep. (Unfortunately.)

File: INSTALL.md (hunk 3 above)
Comment (neongreen) [reply INSTALL.md]: Code. It's created in `Guide.Config.readConfig` (not very intuitively).

File: README.md (hunk 4 above)
Comment (neongreen) [reply README.md:50]: We don't currently have a description of the frontend structure, I will ask the frontend team to have something written in Notion once things slow down (and then we can link the Notion page in the repo).

**Review by willbasky — APPROVED** [review]

**Pull request merged**